dev/ProgressRing/ProgressRingAutomationPeer.cpp

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "ResourceAccessor.h"
+#include "ProgressRing.h"
+#include "ProgressRingAutomationPeer.h"
+#include "Utils.h"
+
+#include "ProgressRingAutomationPeer.properties.cpp"
+
+ProgressRingAutomationPeer::ProgressRingAutomationPeer(winrt::ProgressRing const& owner)
+    : ReferenceTracker(owner)
+{
+}
+
+// IAutomationPeerOverrides
+winrt::IInspectable ProgressRingAutomationPeer::GetPatternCore(winrt::PatternInterface const& patternInterface)
+{
+    if (patternInterface == winrt::PatternInterface::RangeValue)
+    {
+        if (auto progressRing = Owner().try_as<winrt::ProgressRing>())
+        {
+            if (progressRing.IsIndeterminate())
+            {
+                return nullptr;
+            }
+        }
+
+        return *this;
+    }
```

Comment:
Lets not try to make this change since there is an issue with Narrator reacting to changing this pattern at runtime. (Spec bug filed on Narrator). You can add a comment here about the issue. The problem is that Narrator caches the patterns and does not react to the list of patterns changing at runtime. Which means if we start in indeterminate state and then switch to determinate, we will end up not reading the percentages which sounds worse than reading the 0% in indeterminate case. (which we have lived with all along)